iceoryx2/src/port/publisher.rs

```diff
@@ -555,11 +554,16 @@ impl<Service: service::Service> PublisherBackend<Service> {
                 let history = unsafe { &mut *history.get() };
                 for i in 0..history.len() {
```

Comment:
Wouldn't it be better to deliver only the number of samples which fit into the queue?
In iceoryx classic we did exactly that.
```rs
let queue_size = ???;
let begin = history.len().saturating_sub(queue_size);
for i in begin..history.len() {
    ...
}
```
If `history` is a `Vec`, one can also just get the slice over the range and iterate that slice. This would also lead to remove an unsafe block.